Ok I removed the brew versions and now I get

```
dom@MacBook-Pro sundials-C_serial-example % make
make
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_analytic.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_analytic_nonlin.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator_fp.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator1D.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_heat1D.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_heat1D_adapt.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_KrylovDemo_prec.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_robertson.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_robertson_constraints.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_robertson_root.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_onewaycouple_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_twowaycouple_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_reaction_diffusion_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_brusselator_1D_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -O3 -DNDEBUG -O3 -DNDEBUG -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -c ark_kpr_mri.c
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_analytic ark_analytic.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_analytic_nonlin ark_analytic_nonlin.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator ark_brusselator.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator_fp ark_brusselator_fp.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator1D ark_brusselator1D.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_heat1D ark_heat1D.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_heat1D_adapt ark_heat1D_adapt.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_KrylovDemo_prec ark_KrylovDemo_prec.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_robertson ark_robertson.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_robertson_constraints ark_robertson_constraints.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_robertson_root ark_robertson_root.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator_mri ark_brusselator_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_onewaycouple_mri ark_onewaycouple_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_twowaycouple_mri ark_twowaycouple_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_reaction_diffusion_mri ark_reaction_diffusion_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_brusselator_1D_mri ark_brusselator_1D_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
/nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang -o ark_kpr_mri ark_kpr_mri.o  -O3 -DNDEBUG  -I/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include -I/nix/store/9qgpqsikqlg89w9i1hxsmv96rf7475gq-suitesparse-5.8.1-dev/include -L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib -lsundials_arkode -lsundials_nvecserial -lsundials_nvecmanyvector -lm -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
ld: warning: directory not found for option '-L/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib'
ld: library not found for -lsundials_arkode
clang-7: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [examples] Error 1
```

`Makefile` contains

```
prefix       = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1
exec_prefix  = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1
includedir   = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include
libdir       = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib

CPP      = /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang
CPPFLAGS = -O3 -DNDEBUG
CC       = /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang
CFLAGS   = -O3 -DNDEBUG
LDFLAGS  = 
LIBS     =  -lm

LINKFLAGS = -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
```

I changed this to

```
prefix       = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1
exec_prefix  = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1
includedir   = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/include
libdir       = /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib

CPP      = /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang
CPPFLAGS = -O3 -DNDEBUG
CC       = /nix/store/511qa169fbxj18b3k2bjwxffqzxw0fag-clang-wrapper-7.1.0/bin/clang
CFLAGS   = -O3 -DNDEBUG
LDFLAGS  = 
LIBS     =  -lm

LINKFLAGS = -Wl,-rpath,/nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib
```

and now everything builds but I get runtime errors such as

```
dom@MacBook-Pro sundials-C_serial-example % ./ark_analytic
./ark_analytic
dyld: Library not loaded: /nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1//nix/store/3gmnpv60l1qnvkgh6hd4fis7pqd644q7-sundials-5.6.1/lib/libsundials_arkode.4.dylib
  Referenced from: /Users/dom/sundials-C_serial-example/./ark_analytic
  Reason: image not found
zsh: abort      ./ark_analytic
```

--- a/pkgs/development/libraries/sundials/default.nix
+++ b/pkgs/development/libraries/sundials/default.nix
@@ -15,6 +15,24 @@ stdenv.mkDerivation rec {
   pname = "sundials";
   version = "5.6.1";
 
+  outputs = [ "out" "examples" ];
+
+  src = fetchurl {
+    url = "https://computation.llnl.gov/projects/${pname}/download/${pname}-${version}.tar.gz";
+    sha256 = "Frd5mex+fyFXqh0Eyh3kojccqBUOBW0klR0MWJZvKoM=";
+  };
+
+  patches = [
+    # Fixing an upstream regression in treating cmake prefix directories:
+    # https://github.com/LLNL/sundials/pull/58
+    (fetchpatch {
+      url = "https://github.com/LLNL/sundials/commit/dd32ff9baa05618f36e44aadb420bbae4236ea1e.patch";
+      sha256 = "kToAuma+2iHFyL1v/l29F3+nug4AdK5cPG6IcXv2afc=";
+    })
+  ];
+
+  nativeBuildInputs = [ cmake ];
+
   buildInputs = [
     python
   ]
@@ -30,22 +48,6 @@ stdenv.mkDerivation rec {
   # [INSTALL_GUIDE.pdf](https://raw.githubusercontent.com/LLNL/sundials/master/INSTALL_GUIDE.pdf)
   ++ stdenv.lib.optionals (kluSupport) [
     suitesparse
-  ];
-  outputs = [ "out" "examples" ];
-
-  nativeBuildInputs = [ cmake ];
-
-  src = fetchurl {
-    url = "https://computation.llnl.gov/projects/${pname}/download/${pname}-${version}.tar.gz";
-    sha256 = "Frd5mex+fyFXqh0Eyh3kojccqBUOBW0klR0MWJZvKoM=";
-  };
-  patches = [
-    # Fixing an upstream regression in treating cmake prefix directories:
-    # https://github.com/LLNL/sundials/pull/58
-    (fetchpatch {
-      url = "https://github.com/LLNL/sundials/commit/dd32ff9baa05618f36e44aadb420bbae4236ea1e.patch";
-      sha256 = "kToAuma+2iHFyL1v/l29F3+nug4AdK5cPG6IcXv2afc=";
-    })
   ];
 
   cmakeFlags = [

--- a/pkgs/development/libraries/sundials/default.nix
+++ b/pkgs/development/libraries/sundials/default.nix
@@ -1,6 +1,7 @@
 { stdenv
 , cmake
 , fetchurl
+, fetchpatch
 , python
 , blas
 , lapack
@@ -12,7 +13,7 @@
 
 stdenv.mkDerivation rec {
   pname = "sundials";
-  version = "5.3.0";
+  version = "5.6.1";
 
   buildInputs = [
     python
@@ -35,32 +36,35 @@ stdenv.mkDerivation rec {
 
   src = fetchurl {
     url = "https://computation.llnl.gov/projects/${pname}/download/${pname}-${version}.tar.gz";
-    sha256 = "19xwi7pz35s2nqgldm6r0jl2k0bs36zhbpnmmzc56s1n3bhzgpw8";
+    sha256 = "Frd5mex+fyFXqh0Eyh3kojccqBUOBW0klR0MWJZvKoM=";
   };
-
   patches = [
-    (fetchurl {
-      # https://github.com/LLNL/sundials/pull/19
-      url = "https://github.com/LLNL/sundials/commit/1350421eab6c5ab479de5eccf6af2dcad1eddf30.patch";
-      sha256 = "0g67lixp9m85fqpb9rzz1hl1z8ibdg0ldwq5z6flj5zl8a7cw52l";
+    # Fixing an upstream regression in treating cmake prefix directories:
+    # https://github.com/LLNL/sundials/pull/58
+    (fetchpatch {
+      url = "https://github.com/LLNL/sundials/commit/dd32ff9baa05618f36e44aadb420bbae4236ea1e.patch";
+      sha256 = "kToAuma+2iHFyL1v/l29F3+nug4AdK5cPG6IcXv2afc=";
     })
   ];
 
   cmakeFlags = [
     "-DEXAMPLES_INSTALL_PATH=${placeholder "out"}/share/examples"
   ] ++ stdenv.lib.optionals (lapackSupport) [
-    "-DLAPACK_ENABLE=ON"
+    "-DENABLE_LAPACK=ON"
     "-DLAPACK_LIBRARIES=${lapack}/lib/liblapack${stdenv.hostPlatform.extensions.sharedLibrary}"
   ] ++ stdenv.lib.optionals (kluSupport) [
-    "-DKLU_ENABLE=ON"
+    "-DENABLE_KLU=ON"
     "-DKLU_INCLUDE_DIR=${suitesparse.dev}/include"
     "-DKLU_LIBRARY_DIR=${suitesparse}/lib"
-  ] ++ stdenv.lib.optionals (lapackSupport && !lapack.isILP64) [
-    # Use the correct index type according to lapack which is supposed to be
-    # the same index type compatible with blas, thanks to the assertion of
-    # buildInputs
-    "-DSUNDIALS_INDEX_TYPE=int32_t"
-  ]
+  ] ++ [(
+    # Use the correct index type according to lapack and blas used. They are
+    # already supposed to be compatible but we check both for extra safety. 64
+    # should be the default but we prefer to be explicit, for extra safety.
+    if blas.isILP64 then
+      "-DSUNDIALS_INDEX_SIZE=64"
+    else
+      "-DSUNDIALS_INDEX_SIZE=32"
+  )]
   ;
 
   doCheck = true;

--- a/pkgs/development/libraries/sundials/default.nix
+++ b/pkgs/development/libraries/sundials/default.nix
@@ -31,6 +31,7 @@ stdenv.mkDerivation rec {
   ++ stdenv.lib.optionals (kluSupport) [
     suitesparse
   ];
+  outputs = [ "out" "examples" ];
 
   nativeBuildInputs = [ cmake ];
 
@@ -48,7 +49,7 @@ stdenv.mkDerivation rec {
   ];
 
   cmakeFlags = [
-    "-DEXAMPLES_INSTALL_PATH=${placeholder "out"}/share/examples"
+    "-DEXAMPLES_INSTALL_PATH=${placeholder "examples"}/share/examples"
   ] ++ stdenv.lib.optionals (lapackSupport) [
     "-DENABLE_LAPACK=ON"
     "-DLAPACK_LIBRARIES=${lapack}/lib/liblapack${stdenv.hostPlatform.extensions.sharedLibrary}"

